CMakeLists.txt

```diff
@@ -920,7 +920,8 @@ target_link_libraries(
   ${OPENSSL_CRYPTO_LIBRARY}
   ${CMAKE_DL_LIBS}
   rt
-  Threads::Threads
+  curl
+  ${CMAKE_THREAD_LIBS_INIT}
```

Comment:
I'm pretty sure @SwooshyCueb changed this to use `Threads::Threads`. It's probably a good idea to get his thoughts on the use of `Threads::Threads` vs `${CMAKE_THREAD_LIBS_INIT}`.

Reply:
Yeah the best practice is to use the imported target if the CMake find module defines one. From the [CMake docs](https://cmake.org/cmake/help/v3.11/module/FindThreads.html):
> Use of both the imported target as well as [the `THREADS_PREFER_PTHREAD_FLAG`] switch is highly recommended for new code.

Reply:
Additionally, we should be calling `find_package(curl REQUIRED)` and using `${CURL_LIBRARIES}` and `${CURL_INCLUDE_DIRS}` instead of just sticking `curl` in the `target_link_libraries` call.
https://cmake.org/cmake/help/v3.11/module/FindCURL.html

Reply:
Do I need to make changes or is this conversation resolved?

Reply:
Two sets of changes need to be made:
- `${CMAKE_THREAD_LIBS_INIT}` needs to be `Threads::Threads`. There's at least one other place you're using `${CMAKE_THREAD_LIBS_INIT}`, so make the change there, too.
- Use the `FindCURL` CMake module.
    - Up towards the top of the file, there are a couple of `find_package` calls. Add another one like `find_package(CURL REQUIRED)`
    - Instead of using `curl` in the `target_link_libraries` calls, use `${CURL_LIBRARIES}` instead.
    - Wherever you're linking to curl, also add `${CURL_INCLUDE_DIRS}` to the `target_include_directories` call.